File: public/exam_list/java/exercises/concept/log-levels/src/main/java/LogLevels.java
import java.util.Scanner;

class Main {
    public static String message(String logLine) {
        throw new UnsupportedOperationException("Please implement the (static) LogLine.message() method");
    }

    public static String logLevel(String logLine) {
        throw new UnsupportedOperationException("Please implement the (static) LogLine.logLevel() method");
    }

    public static String reformat(String logLine) {
        throw new UnsupportedOperationException("Please implement the (static) LogLine.reformat() method");
    }

    public static void main(String[] args) {
        Scanner reader = new Scanner(System.in);
        String str = reader.nextLine();
        
        System.out.println(message(str));
        System.out.println(logLevel(str));
        System.out.println(reformat(str));
    }
}

File: public/exam_list/cpp/exercises/concept/log-levels/log_levels.cpp
#include <string>
#include <iostream>

using namespace std;

string message(const string& line) { return line.substr(line.find(':') + 2); }

string log_level(const string& line) {
    return line.substr(1, line.find(']') - 1);
}

string reformat(const string& line) {
    return message(line) + " (" + log_level(line) + ')';
}

int main() {
    string str;
    cin >> str;

    cout << log_level(str) << endl << message(str) << endl << reformat(str);
    return 0;
}